toolchain/base/runtime_sources.bzl

```
@@ -0,0 +1,435 @@
+        BUILTINS_AARCH64_SRCS,
+        BUILTINS_X86_64_SRCS,
+        BUILTINS_I386_SRCS,
+    ]
+    for src in srcs
+]
+
+#    "aarch64-unknown-linux-gnu": BUILTINS_GENERIC_SRCS + BUILTINS_BF16_SRCS + BUILTINS_TF_SRCS
+#    "x86_64-unknown-linux-gnu": BUILTINS_GENERIC_SRCS + BUILTINS_BF16_SRCS + BUILTINS_TF_SRCS + BUILTINS_X86_ARCH_SRCS + BUILTINS_X86_FP80_SRCS + [
+#    "i386-unknown-linux-gnu": BUILTINS_GENERIC_SRCS + BUILTINS_BF16_SRCS + BUILTINS_X86_ARCH_SRCS + BUILTINS_X86_FP80_SRCS + [
+
+#[carbon_cc_runtime_static_library(
+#    name = target + "/libclang_rt.builtins.a",
+#    srcs = ["@llvm-project//compiler-rt:lib/builtins/" + src for src in BUILTINS_TARGET_SRCS[target]],
+#    copts = [
+#        "--target=" + target,
+#        "-std=c11",
+#        "-fPIC",
+#        "-ffreestanding",
+#        "-fno-builtin",
+#        "-fvisibility=hidden",
+#        "-fomit-frame-pointer",
+#    ],
+#) for target in TARGETS]
+# The main LLVM command line tools, including their "primary" name, binary name,
+# and the library dependency required to use them.
+
+_TEMPLATE = """
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Generated header file of strings describing the runtime library sources
+// needed by various Carbon runtime libraries.
+//
+// See toolchain/driver/runtime_sources.bzl for more details.
+
+#ifndef CARBON_TOOLCHAIN_BASE_RUNTIME_SOURCES_H_
+#define CARBON_TOOLCHAIN_BASE_RUNTIME_SOURCES_H_
+
+#include "llvm/ADT/StringRef.h"
+
+namespace Carbon::RuntimeSources {{
+
+constexpr inline llvm::StringLiteral CrtBegin = {crtbegin};
+constexpr inline llvm::StringLiteral CrtEnd = {crtend};
+
+constexpr inline llvm::StringLiteral BuiltinsGenericSrcs[] = {{
+{generic_srcs}
+}};
+constexpr inline llvm::StringLiteral BuiltinsMacosSrcs[] = {{
+{macos_srcs}
+}};
+constexpr inline llvm::StringLiteral BuiltinsBf16Srcs[] = {{
+{bf16_srcs}
+}};
+constexpr inline llvm::StringLiteral BuiltinsTfSrcs[] = {{
+{tf_srcs}
+}};
+constexpr inline llvm::StringLiteral BuiltinsX86ArchSrcs[] = {{
+{x86_arch_srcs}
+}};
+constexpr inline llvm::StringLiteral BuiltinsX86Fp80Srcs[] = {{
+{x86_fp80_srcs}
+}};
+constexpr inline llvm::StringLiteral BuiltinsAarch64Srcs[] = {{
+{aarch64_srcs}
+}};
+constexpr inline llvm::StringLiteral BuiltinsX86_64Srcs[] = {{
+{x86_64_srcs}
+}};
+constexpr inline llvm::StringLiteral BuiltinsI386Srcs[] = {{
+{i386_srcs}
+}};
+
+}}
+
+#endif  // CARBON_TOOLCHAIN_BASE_RUNTIME_SOURCES_H_
+"""
+
+def _make_srcs_strings(dir, srcs):
+    return "\n".join(['    "' + dir + "/" + src + '",' for src in srcs])
+
+def _generate_runtime_sources_h_rule(ctx):
+    h_file = ctx.actions.declare_file(ctx.label.name)
+    ctx.actions.write(h_file, _TEMPLATE.format(
+        crtbegin = '"builtins/' + BUILTINS_CRTBEGIN + '"',
+        crtend = '"builtins/' + BUILTINS_CRTEND + '"',
```

Similar to the above:
```suggestion
        crtbegin = '"builtins/{0}"'.format(BUILTINS_CRTBEGIN),
        crtend = '"builtins/{0}"'.format(BUILTINS_CRTEND),
```